lib/trino-filesystem/src/main/java/io/trino/filesystem/Location.java

```diff
@@ -178,6 +178,21 @@ public String fileName()
         return path.substring(path.lastIndexOf('/') + 1);
     }
 
+    /**
+     * Returns a new location with the same parent directory as the current location,
+     * but with the filename corresponding to the specified name.
+     * The location must be a valid file location.
+     */
+    public Location sibling(String name)
+    {
+        requireNonNull(name, "name is null");
+        checkArgument(!name.isEmpty(), "name is empty");
+        checkArgument(!name.isBlank(), "name is blank");
```

Comment:
isBlank has no special meaning here.
we either support leading/trailing whitespace or not, regardless whether there is any non-whitespace character in the middle

we probably don't need to support `.sibling(" ")` , but from filesystem location design perspective, this should be valid.

see https://github.com/trinodb/trino/pull/18206